### DISCUSSION TITLE
I implemented a New PowerShell Handler

### DISCUSSION THREAD

**Discussion by L015H4CK** [cmu-sei/GHOSTS#277] [Show and tell]
Hello everyone (and especially Dustin again),

I have been working with GHOSTS for some weeks now and wanted to execute PowerShell commands directly, i.e. not using the Command Handler and commands like `powershell -c (...)`.

So I went ahead and implemented a new PowerShell Handler - see [my fork](https://github.com/L015H4CK/GHOSTS)([direct compare with master](https://github.com/cmu-sei/GHOSTS/compare/master...L015H4CK:GHOSTS:master)). Basically, the PowerShell Handler is implemented the same way as Command Handler, with the only difference that it calls `powershell.exe` instead of `cmd.exe` ([see here](https://github.com/L015H4CK/GHOSTS/blob/master/src/Ghosts.Client/Handlers/PowerShell.cs#L118)).

Additionally, I wanted to be able to use randomness during the PowerShell command execution. Since the BrowserHandler already implemented such randomness, I used the same technique to replace placeholders in the PowerShell commands ([see here](https://github.com/L015H4CK/GHOSTS/blob/master/src/Ghosts.Client/Handlers/PowerShell.cs#L100)). I think it would be nice to implement this functionality for the Command Handler as well.

Also, I will happily write down a few steps on how to implement a new Handler for the documentation (in case I did everything correctly and as intended) so that people can integrate new handlers more easily (because it took me some time to integrate it into the various components of GHOSTS).

I will gladly answer any questions and open a pull request if the PowerShell Handler should be integrated into GHOSTS.

Best regards,
L015 :fairy_man:

**Comment by sei-dupdyke** [comment]
Awesome — I am currently working on and testing version 8, and I will integrate this handler into the windows client. Thanks for the good work and support of this project!

**Comment by L015H4CK** [comment]
I am happy to hear the handler will be integrated! I am looking forward to the release of version 8.

**Reply by sei-dupdyke** [reply]
Integrated with `bfe4ced9263d59337221710cfe847441e28e7672`
